src/TestUtils/CloudFunctionDeploymentTrait.php

```diff
@@ -133,4 +136,47 @@ public function getBaseUri()
     {
         return self::$fn->getBaseUrl(getenv("GOOGLE_SKIP_DEPLOYMENT") === 'true');
     }
+
+    /**
+     * Retrieve and process logs for the defined function.
+     *
+     * @param string $startTime RFC3339 timestamp marking start of time range to retrieve.
+     * @param callable $process callback function to run on the logs.
+     */
+    private function processFunctionLogs(string $startTime, callable $process)
+    {
+        $projectId = getenv('GOOGLE_PROJECT_ID');
+
+        if (empty(self::$loggingClient)) {
+            self::$loggingClient = new LoggingClient([
+                'projectId' => $projectId
+            ]);
+        }
+
+        // Define the log search criteria.
+        $logFullName = 'projects/' . $projectId . '/logs/cloudfunctions.googleapis.com%2Fcloud-functions';
+        $filter = sprintf(
+            'logName="%s" resource.labels.function_name="%s" timestamp>="%s"',
+            $logFullName,
+            self::$fn->getFunctionName(),
+            $startTime
+        );
+
+        echo "\nRetrieving logs [$filter]...\n";
```

Comment:
nit: using "\n" here but PHP_EOL below. I prefer PHP_EOL but really just would like us to be consistent

Reply:
Fixed.